### ISSUE TITLE
Store header is ignored during Customer operations. Only token is taken into account

### ISSUE THREAD

**Issue by TomashKhamlai** [magento/graphql-ce#770]
### Preconditions (*)

1. Two websites

|Web Site|Store|Store View|
|---|---|---|
|Main Website<br>`base`|Main Website Store<br>`main_website_store`|Default Store View<br>`default`|
|Test Website<br>`additional`|Test Website Store<br>`test_website_store`|Test Store View<br>`test`|
2. **Account Sharing Options** has its default global setting **Share Customer Accounts** = **Per Website**

### Steps to reproduce (*)

1. Create Customer on the `default` Store View
**Headers**
```json
{
  "Store": "default"
}
```

**Request**
```graphql
mutation {
  createCustomer(
    input: {
      prefix: "Prince"
      firstname: "John"
      suffix: "IV"
      middlename: "Ruel"
      lastname: "Doe"
      email: "John.Doe@example.com"
      dob: "10/10/10"
      gender: 1
      password: "123123qA"
    }
  ) {
    customer {
      prefix
      firstname
      lastname
      middlename
      suffix
      email
      dob
      gender
    }
  }
}
```
2. Create Customer on the `test` Store View
**Headers**
```json
{
  "Store": "test"
}
```

**Request**
```graphql
mutation {
  createCustomer(
    input: {
      prefix: "Prince"
      firstname: "John"
      suffix: "IV"
      lastname: "Doe"
      email: "John.Doe@example.com"
      dob: "11/11/11"
      gender: 3
      password: "123123qAqA"
    }
  ) {
    customer {
      prefix
      firstname
      lastname
      middlename
      suffix
      email
      dob
      gender
    }
  }
}
```
2. Generate Customer's token for the customer on `default` Store View
**Headers**
```graphql
{
  "Store": "default"
}
```
**Request**
```graphql
mutation {
  generateCustomerToken(
    email: "John.Doe@example.com"
    password: "123123qA"
  ) {
    token
  }
}
```
3. Get Customers information from the `test` Store View using token for `default` Store Virew
**Headers**
{
  "Authorization": "Bearer xo0aqut1x5erzjv2du6vu6qbbegkjpmr",
  "Store": "ssetview"
}
**Request**
```graphql
query {
  customer {
    created_at
    default_billing
    default_shipping
    dob
    email
    firstname
    gender
    group_id
    id
    is_subscribed
    lastname
    middlename
    prefix
    suffix
    taxvat
    group_id
  }
}
```
### Expected result (*)

1. Authorization problem

### Actual result (*)

1. Information was returned from the default Store View.

**Comment by lenaorobei** [comment]
Checks like `is customer locked` or `scope` should happen before customer will even get a token.

**Comment by lenaorobei** [comment]
@TomashKhamlai could you please create separate issue in the https://github.com/magento/magento2 repo because it is not GraphQL issue?

**Comment by lenaorobei** [comment]
And close this one with reference https://github.com/magento/graphql-ce/issues/750

**Comment by TomashKhamlai** [comment]
Closed #750
